Add user documentation for gvm-tools (#163)
greenbone/gvm-tools: docs/config.rst

.. _config:

Configuration
=============

.. versionchanged:: 2.0

> By default :program:`gvm-tools` :ref:`programs <tools>` are evaluating the

Reviewer: ```suggestion
By default, :program:`gvm-tools` :ref:`programs <tools>` are evaluating the
```

:file:`~/.config/gvm-tools.conf`
`ini style <https://docs.python.org/3/library/configparser.html#supported-ini-file-structure>`_

> config file since version 2.0. The read config file can be adjusted with the

Reviewer: ```suggestion
config file since version 2.0. The name of the config file to be used can be set with the
```

:command:`-c/--config` command line switch.

Settings
--------

The configuration file consists of sections, each led by a :code:`[section]`
header, followed by key/value entries separated by a :code:`=` character.
Whitespace between key and value is ignored. Meaning :code:`key = value` is the
same as :code:`key=value`.

Currently five sections are evaluated:

* :ref:`Main section <main_section>`
* :ref:`GMP section <gmp_section>`
* :ref:`Socket section <socket_section>`
* :ref:`TLS section <tls_section>`
* :ref:`SSH section <ssh_section>`

.. _main_section:

.. rubric:: Main section

> The main section allows to change the default connection timeout besides

Reviewer: ```suggestion
The main section allows changing the default connection timeout besides
```

defining variables for :ref:`interpolation`.

.. code-block:: ini

  [main]
  timeout = 60

.. _gmp_section:

.. rubric:: GMP section

> The GMP section allows for setting the default username and password for

Reviewer: ```suggestion
The GMP section allows setting the default username and password for
```

`GMP (Greenbone Management Protocol)
<https://community.greenbone.net/t/about-the-greenbone-management-protocol-gmp-category/83>`_
based communication.

.. code-block:: ini

  [gmp]
  username=gmpuser
  password=gmppassword

.. _socket_section:

.. rubric:: Socket section

> The socket section allows to set the default path to the unix socket of

Reviewer: ```suggestion
The socket section allows setting the default path to the unix socket of
```

> The socket section allows setting the default path to the unix socket of

Reviewer: ```suggestion
The socket section allows setting the default path to the Unix Domain socket of
```

:term:`gvmd` or :term:`openvasmd` respectively. Not to be confused with the
socket path to the redis server used by :term:`openvassd`. Only relevant if
the socket connection type is used.

.. code-block:: ini

  [unixsocket]
  socketpath=/var/run/gvmd.sock

.. _tls_section:

.. rubric:: TLS section

> The TLS section allows to set the default port, TLS certificate file, TLS key

Reviewer: ```suggestion
The TLS section allows setting the default port, TLS certificate file, TLS key
```

file and TLS certificate authority file. Only relevant if the TLS connection
type is used (Default for :term:`GOS` 3.1).

.. code-block:: ini

  [tls]
  port=1234
  certfile=/path/to/tls.cert
  keyfile=/path/to/tls.key
  cafile=/path/to/tls.ca

.. _ssh_section:

.. rubric:: SSH section

> The SSH section allows to set the default SSH port, SSH username and SSH

Reviewer: ```suggestion
The SSH section allows setting the default SSH port, SSH username and SSH
```

password. Only relevant if the SSH connection type is used (Default for
:term:`GOS` 4 and beyond).

.. code-block:: ini

  [ssh]
  username=sshuser
  password=sshpassword
  port=2222

.. rubric:: Comments

Configuration files may also contain comments by using the special character
:code:`#`. A comment should be placed on a separate line above or below the
setting.

.. code-block:: ini

  [main]
  # connection timeout of 120 seconds
  timeout=120

.. _interpolation:

.. rubric:: Interpolation

> The configuration file even supports `interpolation of values

Reviewer: ```suggestion
The configuration file also supports `interpolation of values
```

<https://docs.python.org/3/library/configparser.html#interpolation-of-values>`_.
It is possible to define values in the :code:`[main]` section which can be

> referenced via a :code:`%(<variablename>)s` syntax. Additionally values of the

Reviewer: ```suggestion
referenced via a :code:`%(<variablename>)s` syntax. Additionally, values of the
```
